scaaml/stats/example_iterator.py

```diff
@@ -30,26 +30,23 @@ class ExampleIterator:
 
     def __init__(self,
                  dataset_path: str,
-                 split: Union[None, str] = None,
-                 group: Union[None, int] = None,
-                 part: Union[None, int] = None) -> None:
+                 split: Optional[str] = None,
```

Comment:
Should we use `Optional[Literal[]]` here to reflect the narrowed choice of strings or would it mess too much with the rest of the code?

Reply:
Python3.7 compatibility. We use this code mainly right after dataset capture, but it could be handy to call it from a Python3.7 environment later.